tools/screenshot-test/src/app/screenshot-result.ts

```diff
@@ -0,0 +1,58 @@
+const config = require('../config.json');
+/**
+ * Results for a screenshot test
+ * Has all related information including Travis Job ID, SHA, pull request number.
+ * And the test results for each test.
+ */
+export class ScreenshotResult {
+  /** PR information: the pull request number */
+  prNumber: string;
+  /** PR information: the sha of the pull request commit */
+  sha: string;
+  /** PR information: The travis job ID */
+  travis: string;
+
+  /** The names of the tests that have screenshot results */
+  testNames: string[];
+  /**
+   * Test result: passed or failed. The value can be true if test failed but
+   * PR approved by user
+   */
+  allTestsPassedOrApproved: boolean;
+  /** Test results: passed or failed for each test */
+  testResultsByName: Map<string, boolean> = new Map<string, boolean>();
+  /**
+   * Test approved: whether the test images are copied to goldens.
+   * The value is the date/time of approval of current commit. Null if never approved.
+   */
+  approvedTime: number;
+
+  githubStatus: any;
```

Comment:
Any way for this to have a more specific type?
(potentially even `object` or `{}` if we can't be more specific)